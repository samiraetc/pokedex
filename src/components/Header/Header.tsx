import React from 'react';
import { Link } from 'react-router-dom';
import pokedex from '../../assets/images/pokedex.png'

export const COMPONENT_ID = 'pokedex-header'

const Header = () => {
    return (
        <div className="header-container"  id={COMPONENT_ID}>
            <img src={pokedex} alt="pokedex logo" className='header-image' />
            <ul className="header-menu">
                <li className="header-menu__item">
                    <Link to="/">Home</Link>
                </li>
                <li className="header-menu__item">
                    <Link to="/favorites">Favorites</Link>
                </li>
            </ul>
        </div>
    )
}

export default Header;