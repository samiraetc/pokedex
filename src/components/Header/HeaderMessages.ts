
export const HeaderMessages = {
    HEADER_HOME_TITLE: 'Home',
    HEADER_FAVORITES_TITLE: 'Favorites',
}