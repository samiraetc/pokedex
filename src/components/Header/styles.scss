.header-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    background-color: $pokedex-color-blue-strong;
    padding: $pokedex-spacing-12;

    .header-image {
        width: 250px;
    }

    .header-menu {
        display: flex;
        flex-direction: row;
        list-style: none;
        gap: $pokedex-spacing-20;
        margin: $pokedex-spacing-12;
    }

    .header-menu__item {
        font-size: $pokedex-font-size-18;
    }

    .header-menu__item a {
        color: $pokedex-color-white;
        text-decoration: none;
    }
}