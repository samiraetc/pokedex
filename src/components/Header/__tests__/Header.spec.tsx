import { render, configure } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom';
import Header, { COMPONENT_ID } from '../Header';

configure({
    testIdAttribute: 'id',
  });
  
describe('[HEADER]', () => {
    test('should render Header', () => {
        const { asFragment, getByText, getByTestId } = render(<Header />, {wrapper: BrowserRouter});

        expect(getByText('Home'));
        expect(getByTestId(COMPONENT_ID)).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });
});