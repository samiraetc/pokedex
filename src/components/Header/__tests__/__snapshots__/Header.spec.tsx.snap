// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[HEADER] should render Header 1`] = `
<DocumentFragment>
  <div
    class="header-container"
    id="pokedex-header"
  >
    <img
      alt="pokedex logo"
      class="header-image"
      src="default"
    />
    <ul
      class="header-menu"
    >
      <li
        class="header-menu__item"
      >
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="header-menu__item"
      >
        <a
          href="/favorites"
        >
          Favorites
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
